Build/Build.cs

```diff
@@ -258,11 +272,36 @@ class Build : NukeBuild
         });
 
     Target SpellCheck => _ => _
+        .OnlyWhenDynamic(() => DocumentationChangesDetected())
         .Executes(() =>
         {
             Node($"{YarnCli} install --silent", workingDirectory: RootDirectory);
             Node($"{YarnCli} --silent run cspell --no-summary", workingDirectory: RootDirectory,
                 customLogger: (_, msg) => Error(msg));
         });
+
+    bool DocumentationChangesDetected()
+    {
+        return Changes.Any(x => x.StartsWith("docs"));
+    }
+
+    bool SourceChangesDetected()
+    {
+        return Changes.Any(x => !x.StartsWith("docs"));
+    }
+
+    IEnumerable<string> Changes
```

Comment:
🔧 You're returning a lazily executed query here, so let's make it return an array instead and change the two methods into properties
🔧 The chained call was a bit misleading because of the newlines and the braces.

```
    string[] Changes
    {
        get
        {
            using var repo = new Repository(GitRepository.LocalDirectory);

            Tree targetBranch = repo.Branches["develop"].Tip.Tree;
            Tree workingDir = repo.Branches[repo.Head.FriendlyName].Tip.Tree;

            return repo.Diff
                .Compare<TreeChanges>(targetBranch, workingDir)
                .Where(x => x.Exists)
                .Select(x => x.Path)
                .ToArray();
        }
    }
```